lib/web/join_tokens.go

```diff
@@ -292,7 +292,13 @@ func getJoinScript(ctx context.Context, settings scriptSettings, m nodeAPIGetter
 	}
 
 	version := proxyServers[0].GetTeleportVersion()
-	hostname, portStr, err := utils.SplitHostPort(proxyServers[0].GetPublicAddr())
+
+	publicAddr := proxyServers[0].GetPublicAddr()
+	if publicAddr == "" {
+		return "", trace.Errorf("proxy public_addr is not set, you must set proxy_service.public_addr to the publicly reachable address of the proxy before you can generate a node join script")
+	}
+
+	hostname, portStr, err := utils.SplitHostPort(publicAddr)
```

Comment:
can we also log something helpful when public_addr is set to just a hostname with no port?

Reply:
I tested and it looks like when public_addr is set without a port, the port defaults to `3080` (set here https://github.com/gravitational/teleport/blob/master/lib/config/configuration.go#L921):

`public_addr: localhost`: the script is generated with port 3080 (and if you log `publicAddr, it is `localhost:3080`)

missing `public_addr`: error

So it looks like logging when the port is missing is not necessary.